core/services/runs.go

```diff
@@ -335,18 +339,20 @@ func validateMinimumConfirmations(
 }
 
 func updateTaskRunConfirmations(currentHeight *models.Big, jr *models.JobRun, taskRun *models.TaskRun) {
-	if jr.CreationHeight == nil || currentHeight == nil {
+	if !taskRun.MinimumConfirmations.Valid || jr.CreationHeight == nil || currentHeight == nil {
```

Comment:
This line seems ripe for a cleanup, but thinking it should hold out for another PR. At its core, we want to more confident in our code path that we are resuming a runlog initiated run because of a new head.